### PR TITLE
unquote in formulas

### PR DESCRIPTION
**Pull request by romainfrancois**
see #4287 and #4199 

``` r
library(dplyr, warn.conflicts = FALSE)

df <- tibble::tibble(year = seq(2015, 2050, 5), P = 5.0 + 2.5 * year)
year <- 2037

summarise_at(df, vars(-year), ~approx(x = year, y = ., xout = !!year)$y)
#> # A tibble: 1 x 1
#>       P
#>   <dbl>
#> 1 5098.

df <- tibble(a = 1:10, b = runif(1:10), c = letters[1:10])
varname <- "a"
symname <- rlang::sym(varname)
quoname <- enquo(symname)

df %>% mutate_at(vars(matches("b")), list(~mean(!!quoname)))
#> # A tibble: 10 x 3
#>        a     b c    
#>    <int> <dbl> <chr>
#>  1     1   5.5 a    
#>  2     2   5.5 b    
#>  3     3   5.5 c    
#>  4     4   5.5 d    
#>  5     5   5.5 e    
#>  6     6   5.5 f    
#>  7     7   5.5 g    
#>  8     8   5.5 h    
#>  9     9   5.5 i    
#> 10    10   5.5 j
```

<sup>Created on 2019-03-22 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

### CODE REVIEW

**Comment by romainfrancois** [comment]
It's possible this can be simplified further, it was doing all sorts of gymnastics to maintain the names of the functions, supposedly for hybrid purposes, but actually the new hybrid prefers when the function called is inlined. , so it does not need to resolve the symbol, and match it against known functions.

**Comment by romainfrancois** [comment]
Something is not quite right here. I launched a revved check with this branch, and it froze around the ELMER package. Might or might not be related to this package 🤷‍♂️ but will need to more carefully review what's happening.

File: R/funs.R [hunk 1]
```diff
@@ -71,38 +71,46 @@ new_funs <- function(funs) {
...
+      if (is_character(.x)) {
+        .x <- get(.x, .env, mode = "function")
+      } else if (!is_function(.x)) {
+        abort("not expecting this")
+      }
+      if (length(args)) {
+        # swoosh the extra arguments so that we have a function that only
+        # takes one argument `.`, or `.x` as a synonym
+        .x <- new_function(
+          list(.=missing_arg(), .x=sym(".")),
```
Comment (lionel-) [inline R/funs.R]:
```suggestion
          list( . =missing_arg(), .x = sym(".")),
```

File: R/funs.R [hunk 2]
```diff
@@ -71,38 +71,46 @@ new_funs <- function(funs) {
...
+  if (!is_character(.funs) && !is_list(.funs)) {
+    .funs <- list(.funs)
   }
 
-  funs <- map(.x, function(fun) {
-    if (is_bare_formula(fun, lhs = FALSE)) {
-      fun <- as_function(fun)
+  funs <- map(.funs, function(.x){
+    if (is_formula(.x)) {
+      .x <- as_function(expr_interp(.x), env = .env)
```
Comment (lionel-) [inline R/funs.R]: Could you pull the `expr_interp()` on its own line? This is an important operation. I missed it at first reading and was confused where interpolation happened.

Comment (lionel-) [reply R/funs.R]: Probably needs a comment that we are processing unquote operators at inlining time but only for rlang lambdas.

**Pull request merged**